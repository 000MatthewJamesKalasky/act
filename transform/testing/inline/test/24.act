deftype foo(int b){}

function baz(foo f) : foo
{
  chp {
    self.b := f.b = 0 ? 0 : 1
  }
}

defproc test ()
{

  foo x;
  foo f;

  chp {
  *[
    x := baz(f)
  ]
  }
}
test t; 
