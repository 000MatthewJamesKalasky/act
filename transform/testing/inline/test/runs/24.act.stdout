deftype foo (int<32> b);
defproc test ();
function baz (foo f) : foo;

deftype foo (int<32> b)
{

/* instances */

/* connections */
}

defproc test ()
{

/* instances */
foo fuse_0;
foo farg_0;
foo f;
foo x;
foo fret_0;

/* connections */
chp {
*[true -> farg_0:=f;fret_0.b:=farg_0.b=0x0 ? 0x0 : 0x1;fuse_0:=fret_0;x:=fuse_0]
}
}

function baz (foo f) : foo
{

/* instances */

/* connections */
chp {
self.b:=f.b=0x0 ? 0x0 : 0x1
}
}


/* instances */
test t;

/* connections */
